sklearn/utils/tests/test_array_api.py

```diff
@@ -692,9 +692,11 @@ def test_sparse_device(csr_container, dispatch):
         raise SkipTest("SCIPY_ARRAY_API is not set: not checking array_api input")
     with config_context(array_api_dispatch=dispatch):
         assert device(a, b) is None
-        assert device(a, numpy.array([1])) is None
+        assert device(a, numpy.array([1])) == ("cpu" if dispatch else None)
```

Comment:
That will fail for numpy<2, right? In this case the numpy array has no `.device` so even when `dispatch is True`, the device will be `None`.

To be honest, the simplest may be to decide you need numpy>=2 for array API.

Right now we have scipy 1.14.0 as the minimum version (released 2024-06-24) for array API and numpy 2.0 was released around the same time (2024-06-16).

Reply:
Yes, and I think this is fine as it is.